lib/compilers/dotnet.ts

```diff
@@ -194,51 +205,157 @@ class DotNetCompiler extends BaseCompiler {
         return compilerResult;
     }
 
+    async publishAot(
+        compiler: string,
+        ilcOptions: string[],
+        ilcSwitches: string[],
+        inputFilename: string,
+        execOptions: ExecutionOptions & {env: Record<string, string>},
+    ): Promise<CompilationResult> {
+        const programDir = path.dirname(inputFilename);
+
+        this.setCompilerExecOptions(execOptions, programDir, true);
+
+        // serialize ['a', 'b', 'c;d', '*e*'] into a=b;c%3B=%2Ae%2A;
+        const msbuildOptions = ilcOptions
+            .map(val => val.replaceAll('*', '%2A').replaceAll(';', '%3B'))
+            .reduce((acc, val, idx) => (idx % 2 === 0 ? (acc ? `${acc}${val}` : `${val}`) : `${acc}=${val};`), '');
+
+        // serialize ['a', 'b', 'c', 'd'] into a;b;c;d
+        const msbuildSwitches = ilcSwitches.join(';');
+
+        const toolVersion = await fs.readFile(`${this.clrBuildDir}/aot/package-version.txt`, 'utf8');
+        const tempFile = `/tmp/ce.dotnet.jitout.tmp.${crypto.randomUUID()}`;
```

Comment:
Unfortunately you can't hardcode paths to `/tmp` here: the files on the host system are not the same files that the compiler will see when running on the (sandboxed) site.

Can you not write this temporary file to the `programDir` ? If not, we may need to come up with another way of doing this: either making a top level directory in the temporary directory somehow, but that would be something we'd need to discuss further (and/or you'd need to hardcode by overriding all the directories elsewhere)